pages/install.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/** @var rex_addon $this */
+
+/* setup process */
+
+if (rex_post('install', 'boolean')) {
+
+    $errors = array();
+
+    // step 1/5: select missing packages we need to download
+    $missingPackages = array();
+    $packages = $this->getProperty('setup')['packages'];
+    if (count($packages) > 0) {
+
+        // fetch list of available packages from to redaxo webservice
+        try {
+            $packagesFromInstaller = rex_install_packages::getAddPackages();
+        } catch (rex_functional_exception $e) {
+            $errors[] = $e->getMessage();
+        }
+
+        foreach ($packages as $id => $fileId) {
+
+            $localPackage = rex_package::get($id);
+            if ($localPackage->isSystemPackage()) {
+                continue;
+            }
+
+            $installerPackage = $packagesFromInstaller[$id]['files'][$fileId];
+            if (!$installerPackage) {
+                $errors[] = $this->i18n('package_not_available', $id);
+            }
+
+            if ($localPackage->getVersion() !== $installerPackage['version']) {
+                $missingPackages[$id] = $fileId;
+            }
+        }
+    }
+
+    // step 2/5: download required packages
+    if (count($missingPackages) > 0 && count($errors) == 0) {
+        foreach ($missingPackages as $id => $fileId) {
+
+            $installerPackage = $packagesFromInstaller[$id]['files'][$fileId];
+            if ($installerPackage) {
+
+                // fetch package
+                try {
+                    $archivefile = rex_install_webservice::getArchive($installerPackage['path']);
+                } catch (rex_functional_exception $e) {
+                    $errors[] = $this->i18n('package_failed_to_download', $id);
```

Comment:
$e mit in die meldung einbauen, oder wenigstens ins syslog schreiben?

Reply:
In den anderen catxh blöcken auch

Reply:
Erledigt, danke!